test/util/log/log.go

```diff
@@ -23,20 +25,32 @@ func New() (*test.Hook, *logrus.Entry) {
 // AssertLoggingOutput compares the logs on `h` with the expected entries in
 // `expected`. It returns a slice of errors encountered, with a zero length if
 // no assertions failed.
+// This function skips audit log entries in `h` to avoid test flakiness. Use
+// audit.AssertAuditingOutput() to verify audit entries.
```

Comment:
this was previously added because all the frontend tests would cause the audit log hooks to fire, even without the new change to the frontend constructor. IIRC, i had to update a number of places with more audit logs to get the tests to pass. my intention is that the frontend tests shouldn't be affected by the audit logs, because those tests aren't written to test audit logs.

AFAICT, changing `testinfra` (e.g., to use an audit null logger?) won't resolve all the broken unit tests, because there are  [tests](https://github.com/Azure/ARO-RP/blob/9210f8f11364b12643b70c687b4d738715cc244e/pkg/frontend/frontend_test.go#L137) that bypass `testinfra`. if `frontend_test.go` is the only spot that asserts the log outcomes, it might not be that bad. I will need to revisit these tests, as this was from a month ago.